### PR TITLE
Display an error message for impossible values of minZoom and maxZoom

### LINKED ISSUE #2264: Display an error message for impossible values of minZoom, zoom, and maxZoom
https://jsfiddle.net/7d365j4s/

In this example,
- `zoom` is greater than `maxZoom`
- `minZoom` is greater than `maxZoom`

**mapbox-gl-js version**: v0.15.0
### Steps to Trigger Behavior

Click on link.
### Expected Behavior

Either error messages of contradictory parameters (preferred), and/or refuse to load the map.
### Actual Behavior

No errors, and map does not zoom.

### PR DESCRIPTION
**Pull request by joswinter**
Fixes #2264

When creating a map where holds that `minZoom > maxZoom` the map is loaded and the user cannot zoom. The changes in this pull request throw an error when the above case holds. See the following example: https://jsfiddle.net/7d365j4s/

## Launch Checklist

The pull request needs to be reviewed by at least two other team members. When the pull request has been reviewed we can close this pull request and create a new pull request to the **mapbox/mapbox-gl-js** repository.

### CODE REVIEW

File: src/geo/transform.js [hunk 1]
```diff
@@ -43,13 +43,17 @@ class Transform {
         if (this._minZoom === zoom) return;
         this._minZoom = zoom;
         this.zoom = Math.max(this.zoom, zoom);
+        if (this._maxZoom != null && this._minZoom > this._maxZoom)
+            throw new Error("maxZoom must always be greater than minZoom");
```
Comment (yappel) [inline src/geo/transform.js]: Mapbox uses Error events to report user errors instead of throwing them. See:
https://github.com/mapbox/mapbox-gl-js/blob/master/CONTRIBUTING.md#code-conventions

File: src/geo/transform.js [hunk 2]
```diff
@@ -23,8 +23,8 @@ class Transform {
         this.tileSize = 512; // constant
 
         this._renderWorldCopies = renderWorldCopies === undefined ? true : renderWorldCopies;
-        this._minZoom = minZoom || 0;
-        this._maxZoom = maxZoom || 22;
+        this.minZoom = minZoom || 0;
+        this.maxZoom = maxZoom || 22;
```
Comment (yappel) [inline src/geo/transform.js:27]: Does get minZoom() still return the correct value, since that function returns this._minZoom not this.minZoom? I may be missing something, but why is this change necessary?

**Review by yappel — CHANGES_REQUESTED** [review]

File: src/geo/transform.js (hunk 2 above)
Comment (joswinter) [reply src/geo/transform.js:27]: Good question. When you use the following declaration `this._minZoom = 1` the value is declared directly but when you use `this.minZoom = 1` the setter of minZoom is called. If I wouldn't have changed this the error wouldn't be thrown when `maxZoom > minZoom` when initializing the map, only when changing the minZoom and maxZoom after the initialization.

Comment (yappel) [reply src/geo/transform.js:27]: Interesting, I did not know that it worked that way in JS.

**Review by yappel — APPROVED** [review]

**Comment by lkrombeen** [comment]
Shouldn't you also add a test for set maxZoom(zoom)?

**Comment by joswinter** [comment]
@lkrombeen I added two additional tests for the maxZoom and minZoom setters.

**Review by lkrombeen — APPROVED** [review]
Approved

**Comment by joswinter** [comment]
Thanks for reviewing! I will now close this pull request and open a pull request to the Mapbox repository.
